js/state/frameStateUtil.js

```diff
@@ -717,7 +745,21 @@ const setTabHoverState = (state, frameKey, hoverState, enablePreviewMode) => {
   return state
 }
 
+const frameLocationMatch = (frame, location) => {
+  if (frame == null) {
+    return false
+  }
+
+  return frame.get('location') === location
```

Comment:
nitpick: we may want to check for `frame.get` before calling it (in case the input is not an immutable.map)